core/src/main/java/org/elasticsearch/index/query/TermsQueryBuilder.java

```diff
@@ -142,72 +283,150 @@ public TermsQueryBuilder disableCoord(boolean disableCoord) {
         return this;
     }
 
+    public boolean disableCoord() {
+        return this.disableCoord;
+    }
+
+    public boolean isTermsLookupQuery() {
+        return this.termsLookup != null;
+    }
+
+    public TermsQueryBuilder termsLookup(TermsLookupBuilder termsLookup) {
+        this.termsLookup = termsLookup;
+        return this;
+    }
+
+    public TermsLookupBuilder termsLookup() {
+        return this.termsLookup;
+    }
+
     /**
      * Sets the index name to lookup the terms from.
      */
     public TermsQueryBuilder lookupIndex(String lookupIndex) {
-        this.lookupIndex = lookupIndex;
+        if (lookupIndex != null) {
```

Comment:
sure I understand that, but it feels weird to handle the case where one calls lookupIndex(null). We still would want to create the underlying lookup probably? then we would validate and throw error if needed. Otherwise it is not a lookup?